src/main/features/offer/form/models/offer.ts

```diff
@@ -17,24 +17,27 @@ export class Offer implements Serializable<Offer> {
   @IsDefined({ message: ValidationErrors.OFFER_REQUIRED })
   @IsNotBlank({ message: ValidationErrors.OFFER_REQUIRED })
   @MaxLength(99000, { message: ValidationErrors.OFFER_TEXT_TOO_LONG })
-  offerText?: string
+  offerText: string
 
   @ValidateNested()
   @IsDefined({ message: ValidationErrors.DATE_REQUIRED })
   @IsValidLocalDate({ message: ValidationErrors.DATE_NOT_VALID })
   @IsFutureDate({ message: ValidationErrors.FUTURE_DATE })
-  completionDate?: LocalDate
+  completionDate: LocalDate
 
-  constructor (offerText?: string, completionDate?: LocalDate) {
-    this.offerText = offerText
-    this.completionDate = completionDate
+  static create (offerText: string, completionDate: LocalDate) {
+    const offer = new Offer()
+    if (offerText || completionDate) {
```

Comment:
What is the purpose of this? Populating fields if any of two arguments is provided looks wired at least. This also in conflict with the facts that both fields are marked as mandatory but here it is possible to instantiate only one field.